### DISCUSSION TITLE
set excludes/includes via mqtt or use prefixes on webinterface

### DISCUSSION THREAD

**Discussion by gnome4711** [ESPresense/ESPresense#688] [General]
Hi,

i tried to exclude id's  by setting them via mqtt (espresense/rooms/[room]/exclude). But it doesn't seem to work. Is the only way to set excluded devices with full ID'S via the webinterface?? I want reduce the network traffic. My environment is really noisy.

Also it would be greate if I could use prefixes by "include-Filtering". I can't use it with full ID because my UUID's from my beacon-simulator on Android-Phones are encrypted and they change every 10 seconds. So all ID's beginning with "iBeacon:...." need to be sent via mqtt.
With Filtering includes I only need to add a few ID's to reduce network traffic

Any ideas??
gnome

**Comment by Beast-Panik** [comment]
I have tried the same but I am not sure about the separators between the IDs. I also tried it via the web interface but with , or ; or space it didn't work. What is the correct separator between the ID's?

**Reply by gnome4711** [reply]
The correct separator between  2 ID's would be a simple space. For example: samsung:771a89d46d89 samsung:6689035748ad

**Reply by Beast-Panik** [reply]
Thank you! That was probably too easy. I will experiment further on the weekend.

**Comment by gnome4711** [comment]
I reflashed one device and all the sudden include filtering with prefixes works. Now it is working on all devices (I reflashed them all then).
I reduced my mqtt traffic around 60%!!

**Comment by noelhibbard** [comment]
The topic needs to be:

espresense/rooms/[room]/exclude/set

Or if you want all rooms to get the same exclude, use this topic:

espresense/rooms/*/exclude/set

Use also have to use retain if you want it to survive device restarts.
